shared/common/newui/modal/index.tsx

```diff
@@ -133,5 +133,11 @@ export function ModalContent({
   className,
   ...props
 }: PropsWithChildren<HTMLAttributes<HTMLDivElement>>) {
-  return <div className={cn(styles.content, className)} {...props} />;
+  return (
+    <div
+      data-testid="payment-form"
```

Comment:
This modal component is used in multiple places, so this attribute for the payment modal shouldn't be added here. Also `props` includes `HTMLAttributes<HTMLDivElement>`, so it should already be possible to do `<ModalContent data-testid="payment-form">...` without any changes to this component.

Reply:
yes yes, this is better... I failed to check how often is this component used